packages/search-ui/src/Events.ts

```diff
@@ -27,14 +29,29 @@ function wireUpEventHandler(handlerName, apiConnector, handlerParam) {
  * an API Connector. The events that are passed directly can also be used
  * to override or proxy handlers provided by API Connectors.
  */
-export default class Events {
+
+type EventOptions = {
+  apiConnector?: APIConnector
+  onSearch?: (searchQuery: any) => void
+  onAutocomplete?: (query: AutocompleteQuery) => void,
+  onResultClick?: (result: any) => void,
+  onAutocompleteResultClick?: (result: any) => void
+} 
+
+class Events {
+
+  private search: () => void;
```

Comment:
this im a little hazy on its use so the types may not be right here. Dropping a note to figure out